src/components/NewMeasure/CreateNewMeasureDialog.test.tsx

```diff
@@ -0,0 +1,283 @@
+
+      const measurementPeriodStartNode = screen.getByTestId(
+        "measurement-period-start"
+      );
+      const measurementPeriodStartInput = within(
+        measurementPeriodStartNode
+      ).getByRole("textbox");
+      userEvent.type(measurementPeriodStartInput, "12/33");
+
+      const measurementPeriodEndNode = screen.getByTestId(
+        "measurement-period-end"
+      );
+      const measurementPeriodEndInput = within(
+        measurementPeriodEndNode
+      ).getByRole("textbox");
+      userEvent.type(
+        measurementPeriodEndInput,
+        mockFormikInfo.measurementPeriodEnd
+      );
+
+      expect(measurementPeriodEndInput.value).toBe(
+        mockFormikInfo.measurementPeriodEnd
+      );
+
+      fireEvent.click(measurementPeriodStartNode);
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("measurementPeriodStart-helper-text")
+        ).not.toBe(null);
+        expect(
+          screen.getByTestId("measurementPeriodStart-helper-text")
+        ).toHaveTextContent("Invalid date format. (mm/dd/yyyy)");
+      });
+
+      const submitButton = await screen.getByTestId(
+        "create-new-measure-save-button"
+      );
+      await waitFor(() => expect(submitButton).toBeDisabled(), {
+        timeout: 3000,
+      });
+    });
+  }, 20000);
+
+  test("checking if error text is displayed when measurement periods are not provided", async () => {
+    await act(async () => {
+      await render(<CreateNewMeasureDialog open={true} onClose={undefined} />);
+
+      const measurementPeriodStartNode = screen.getByTestId(
+        "measurement-period-start"
+      );
+      const measurementPeriodStartInput = within(
+        measurementPeriodStartNode
+      ).getByRole("textbox");
+      userEvent.type(measurementPeriodStartInput, "");
+
+      const measurementPeriodEndNode = screen.getByTestId(
+        "measurement-period-end"
+      );
+      const measurementPeriodEndInput = within(
+        measurementPeriodEndNode
+      ).getByRole("textbox");
+      userEvent.type(
+        measurementPeriodEndInput,
+        mockFormikInfo.measurementPeriodEnd
+      );
+
+      expect(measurementPeriodEndInput.value).toBe(
+        mockFormikInfo.measurementPeriodEnd
+      );
+
+      fireEvent.click(measurementPeriodStartNode);
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("measurementPeriodStart-helper-text")
+        ).not.toBe(null);
+        expect(
+          screen.getByTestId("measurementPeriodStart-helper-text")
+        ).toHaveTextContent("Measurement period start date is required");
+      });
+
+      const submitButton = await screen.getByTestId(
+        "create-new-measure-save-button"
+      );
+      await waitFor(() => expect(submitButton).toBeDisabled(), {
+        timeout: 3000,
+      });
+    });
+  }, 20000);
+
+  test("checking if error text is displayed when measurement periods are not provided", async () => {
```

Comment:
Looks like there are 2 test cases with the same description.

Reply:
I just changed it